packages/react/src/components/RuleBuilder/RuleBuilder.jsx

```diff
@@ -0,0 +1,470 @@
+      setCurrentFilter((prev) => {
+        return {
+          ...prev,
+          filterTags: prev.filterTags.filter((t) => t !== tag),
+        };
+      });
+    },
+    []
+  );
+
+  const editUsers = useMemo(
+    () => currentFilter?.filterAccess?.filter((user) => user.access === 'edit') ?? [],
+    [currentFilter]
+  );
+
+  const readUsers = useMemo(
+    () => currentFilter?.filterAccess?.filter((user) => user.access === 'read') ?? [],
+    [currentFilter]
+  );
+
+  const handleUserSelect = useCallback(
+    (access) => (selectedUsers) => {
+      setModalState((prev) => ({
+        ...prev,
+        isOpen: false,
+      }));
+
+      setCurrentFilter((prev) => ({
+        ...prev,
+        filterAccess: [
+          ...(prev.filterAccess?.filter((user) => user.access !== access) ?? []),
+          ...selectedUsers.map((user) => ({ ...user, access })),
+        ],
+      }));
+    },
+    []
+  );
+
+  return (
+    <section
+      className={classnames(baseClass, className)}
+      id={currentFilter?.id}
+      data-testid={currentFilter?.id || 'rule-builder'}
+    >
+      <header className={`${baseClass}--header`}>
+        <div>
+          <h1 className={`${baseClass}--header-title`}>
+            {currentFilter?.filterTitleText || defaultTitleText}
+          </h1>
+          {currentFilter?.filterMetaText && (
+            <p className={`${baseClass}--header-metatext`}>{currentFilter?.filterMetaText}</p>
+          )}
+        </div>
+        <div className={`${baseClass}--header-actions`}>
+          {actions}
+          <Button
+            className={`${baseClass}--header-actions-save`}
+            data-testid="rule-builder-save"
+            onClick={() => onSave(currentFilter)}
+            size="small"
+          >
+            {mergedI18n.saveLabel}
+          </Button>
+        </div>
+      </header>
+      <div className={`${baseClass}--body`}>
+        <Tabs className={`${baseClass}--tabs`}>
+          <Tab className={`${baseClass}--tab`} label={mergedI18n.filterTabText}>
+            <Editor
+              defaultRules={currentFilter.filterRules}
+              columns={currentFilter.filterColumns}
+              onChange={handleOnChange}
+            />
+          </Tab>
+          <Tab className={`${baseClass}--tab`} label={mergedI18n.sharingTabText}>
+            <Accordion>
+              <AccordionItem title={mergedI18n.detailsAccordionText} open>
+                <TextInput
+                  type="text"
+                  id="rule-builder-title-input"
+                  className={`${baseClass}--title-input`}
+                  labelText={mergedI18n.filterNameLabel}
+                  light
+                  value={currentFilter.filterTitleText}
+                  placeholder="Untitled 01"
+                  onChange={(e) => handleOnChange(e.target.value, 'TITLE')}
+                />
+                <FilterTags
+                  hasOverflow
+                  tagContainer={RuleBuilderTags}
```

Comment:
Passing the forwardRef RuleBuilderTags component here is what required the addition of the `i18n` and `onChange` to the FilterTags component. As mentioned in the PR, this could be pulled into the RuleBuilder component, passed these props, and memoized instead of changing the public api.